router-tests/file_upload_test.go

```diff
@@ -0,0 +1,130 @@
+package integration_test
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router/core"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"net/http"
+	"testing"
+
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+)
+
+func TestSingleFileUpload(t *testing.T) {
+	t.Parallel()
+	testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+		files := make([][]byte, 1)
+		files[0] = []byte("File content as text")
+		res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+			Query:     "mutation ($file: Upload!){singleUpload(file: $file)}",
+			Variables: []byte(`{"file":null}`),
+			Files:     files,
+		})
+		require.JSONEq(t, `{"data":{"singleUpload": true}}`, res.Body)
+	})
+}
+
+func TestSingleFileUpload_InvalidFileFormat(t *testing.T) {
+	t.Parallel()
+	testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+		res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+			Query:     "mutation ($file: Upload!){singleUpload(file: $file)}",
+			Variables: []byte(`{"file":"invalid_format"}`),
+		})
+		require.JSONEq(t, `{"errors":[{"message":"Failed to fetch from Subgraph '0' at Path 'mutation'.","extensions":{"errors":[{"message":"string is not an Upload","path":["singleUpload","file"]}],"statusCode":200}},{"message":"Cannot return null for non-nullable field 'Mutation.singleUpload'.","path":["singleUpload"]}],"data":null}`, res.Body)
+	})
+}
+
+func TestSingleFileUpload_NoFileProvided(t *testing.T) {
+	t.Parallel()
+	testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+		res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+			Query:     "mutation ($file: Upload!){singleUpload(file: $file)}",
+			Variables: []byte(`{"file":null}`),
+		})
+		require.JSONEq(t, `{"errors":[{"message":"Failed to fetch from Subgraph '0' at Path 'mutation'.","extensions":{"errors":[{"message":"cannot be null","path":["variable","file"],"extensions":{"code":"GRAPHQL_VALIDATION_FAILED"}}],"statusCode":422}},{"message":"Cannot return null for non-nullable field 'Mutation.singleUpload'.","path":["singleUpload"]}],"data":null}`, res.Body)
+	})
+}
+
+func TestSingleFileUpload_FileSizeExceedsLimit(t *testing.T) {
+	t.Parallel()
+	testenv.Run(t, &testenv.Config{
+		RouterOptions: []core.Option{core.WithRouterTrafficConfig(&config.RouterTrafficConfiguration{
+			MaxUploadRequestBodyBytes: 100,
+		})},
+	}, func(t *testing.T, xEnv *testenv.Environment) {
+		files := make([][]byte, 1)
+		files[0] = []byte("This is an example of a large file that exceeds the max request body size.")
```

Comment:
Could you assert here that the file is bigger than the limit?

Reply:
I already check the response. The limit is for the entire request body not individual files.

Reply:
ok